### PR TITLE
Update documentation and add test case for data cache setting as it is available for postgres now

### PR DESCRIPTION
**Pull request by nityanavali1**
b/299670813
Now that data cache is supported for PostgreSQL, this PR is for updating documentation and extending test cases for PostgreSQL data cache support

Resource Reference : https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#data_cache_enabled

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
note from rileykarson@: I've made the docs update an enhancement intentionally- since it'll make it clear that new functionality is available through the API, even if there was no code change in Terraform
```

```release-note:enhancement
sql: update `google_sql_database_instance` to reflect that `data_cache_enabled` works for PostgresQL now, in addition to MYSQL
```

### CODE REVIEW

**Comment by rileykarson** [comment]
AlloyDB stuff is certainly unrelated, we had to fix something in a way that triggered that error.

File: mmv1/third_party/terraform/services/sql/resource_sql_database_instance_test.go [hunk 1]
```diff
@@ -1587,6 +1587,27 @@ func TestAccSQLDatabaseInstance_sqlMysqlDataCacheConfig(t *testing.T) {
...
+func TestAccSQLDatabaseInstance_sqlPostgresDataCacheConfig(t *testing.T) {
+	t.Parallel()
+	instanceName := "tf-test-enterprise-plus" + acctest.RandString(t, 10)
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccSqlDatabaseInstanceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testGoogleSqlDatabaseInstance_sqlPostgresDataCacheConfig(instanceName),
```
Comment (isaurabhuttam) [inline mmv1/third_party/terraform/services/sql/resource_sql_database_instance_test.go:1599]: Let's add the check for `data_cache_enabled` property was set or not.

File: mmv1/third_party/terraform/services/sql/resource_sql_database_instance_test.go [hunk 2]
```diff
@@ -1587,6 +1587,27 @@ func TestAccSQLDatabaseInstance_sqlMysqlDataCacheConfig(t *testing.T) {
 	})
 }
 
+func TestAccSQLDatabaseInstance_sqlPostgresDataCacheConfig(t *testing.T) {
```
Comment (isaurabhuttam) [inline mmv1/third_party/terraform/services/sql/resource_sql_database_instance_test.go:1590]: Let's add one failure test case for enterprise edition.

Comment (nityanavali1) [reply mmv1/third_party/terraform/services/sql/resource_sql_database_instance_test.go:1590]: Done

File: mmv1/third_party/terraform/services/sql/resource_sql_database_instance_test.go (hunk 1 above)
Comment (nityanavali1) [reply mmv1/third_party/terraform/services/sql/resource_sql_database_instance_test.go]: Done

**Comment by nityanavali1** [comment]
All tests are passing except TestAccAlloydbCluster_restore (https://screenshot.googleplex.com/9ibB34RH3TVL7Xi) (failed because of not enough resources available in us-central1). This test is not related to this change. Can I get approval for the merge. As this is needed for GA launch of Postgres with Data cache feature.

**Comment by rileykarson** [comment]
Note: I cleared cassettes other than TestAccSQLDatabaseInstance_sqlPostgresDataCacheConfig, so that failure cassette won't be merged @ merge time.

**Review by rileykarson — APPROVED** [review]

**Pull request merged**